ledger/testchain-generator/src/main.rs

```diff
@@ -99,8 +140,8 @@ fn generate_testchain<N: Network>(args: Args) -> Result<()> {
         let batch_size = (num_blocks - blocks.len()).min(100);
         let mut batch = builder.generate_blocks(batch_size, &mut rng).with_context(|| "Failed to generate blocks")?;
 
-        println!("Generated {pos} of {num_blocks} blocks");
         pos += batch_size;
+        println!("Generated {pos} of {num_blocks} blocks");
```

Comment:
This is a drive-by fix; without it, `pos` is always outdated by a single iteration.